packages/api/src/client/types/app/bsky/system/declaration.ts

```diff
@@ -1,12 +1,7 @@
 /**
 * GENERATED CODE - DO NOT MODIFY
 */
-export type ActorKnown =
-  | 'app.bsky.system.actorUser'
-  | 'app.bsky.system.actorScene'
-export type ActorUnknown = string
-
 export interface Record {
-  actorType: ActorKnown | ActorUnknown;
+  actorType: string;
```

Comment:
There's a nice TS trick we can use to get the `knownValues` into the type system.  The types validate the same way, but the allowed values are preserved and can appear e.g. through intellisense:
```js
  actorType: 'app.bsky.system.actorUser' | 'app.bsky.system.actorScene' | (string & {});
```

Reply:
Typescript will upcast `'my-string' | string` to `string`, and you lose the information in the type system that `'my-string'` is a special value.  Here's an example:
<img width="346" alt="Screen Shot 2022-12-01 at 11 25 37 AM" src="https://user-images.githubusercontent.com/3150233/205106171-519b8d4d-023a-41cf-89a9-3dbedd0043ea.png">
<img width="393" alt="Screen Shot 2022-12-01 at 11 25 25 AM" src="https://user-images.githubusercontent.com/3150233/205106175-5631b208-d482-48e5-b632-aecba7b5d35c.png">

Reply:
Interesting! Okay I'll do that when I add the feature to codegen.

Reply:
Done!